### PR TITLE
clims-289, Fix MultiFormatFile taking path as input

### PR DESCRIPTION
**Pull request by GitEdvard**
A similar class was defined within the sample sub validation module. I
wanted a more central place for it, since it now was used at two
locations.

### CODE REVIEW

**Comment by johandahlberg** [comment]
I'm not sure we actually need two classes for this. Wouldn't it be better to do a single class, and then have several factory methods to initiate it? This could either take the form of a `MultiFormatFileFactor` class, or which I think is a little bit more pythonic, just two functions, one `from_organization_file(...)` and one `from_path(...)`. What do you think?

**Comment by GitEdvard** [comment]
I have now added a wrapper class for MultiFormatFile, having factory methods from_path and from_organziation_file. I'm not really sure this is what you had in mind though. I was reluctant to merge it in one single class, because it would require an if-else clause in each method, and I didn't like that. I may still change my mind if you insist, so please speak your opinion!

**Comment by GitEdvard** [comment]
Now, there is a refactored version of the MultiFormatFile. I have tried to avoid having multiple calls to xl. open_workbook spread out over two classes. Instead I have created internal classes of type FileContext, which deal with the temp file for organization files.

Note that tests in the snpseq repo will fail until the corresponding PR there will be merged.

File: src/clims/models/file.py [hunk 1]
```diff
@@ -1,53 +1,129 @@
...
-                f.write(chunk)
+    @classmethod
+    def from_path(cls, file_path):
+        file_context = HarddiskFileContext(file_path)
+        return MultiFormatFile(file_context=file_context)
+
+    @classmethod
+    def from_organization_file(cls, organization_file):
+        file_context = OrganizationFileContext(organization_file)
+        return MultiFormatFile(file_context=file_context)
```
Comment (johandahlberg) [inline src/clims/models/file.py]: I think these two should be made `staticmethod` rather than `classmethod`.

**Review by johandahlberg — CHANGES_REQUESTED** [review]

**Pull request merged**